### PR TITLE
Use writable buffer writer

### PR DESCRIPTION
**Pull request by pakrym**
/cc @benaadams

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.Kestrel/Adapter/Internal/StreamSocketOutput.cs [hunk 1]
```diff
@@ -42,18 +42,18 @@ public void Write(ArraySegment<byte> buffer, bool chunk)
...
-                        ChunkWriter.WriteEndChunkBytes(ref writableBuffer);
+                        var writer = new WritableBufferWriter(writableBuffer);
+                        ChunkWriter.WriteBeginChunkBytes(ref writer, buffer.Count);
+                        writer.Write(buffer.Array, buffer.Offset, buffer.Count);
+                        ChunkWriter.WriteEndChunkBytes(ref writer);
                     }
                     else
                     {
-                        writableBuffer.WriteFast(buffer);
+                        writableBuffer.Write(buffer);
```
Comment (pakrym) [inline src/Microsoft.AspNetCore.Server.Kestrel/Adapter/Internal/StreamSocketOutput.cs]: Would be replaced by FastWrite implementation from https://github.com/dotnet/corefxlab/pull/1387

**Comment by davidfowl** [comment]
@pakrym run a measurement on the end to end. Last I checked it didn't improve anything.

**Comment by pakrym** [comment]
@davidfowl what if it improves Output benchmark only?

**Comment by davidfowl** [comment]
This one did too https://github.com/aspnet/KestrelHttpServer/pull/1562.

**Comment by davidfowl** [comment]
Run both and report the numbers.

**Comment by davidfowl** [comment]
@pakrym builds breaks

**Comment by pakrym** [comment]
```
dev

[01:07:53.530] Warmup
[01:08:09.631] RPS: 1197771.77
[01:08:09.633] Benchmark
[01:08:25.671] RPS: 1225775.69

KestrelHttpServer@pakrym/wbw

[01:09:36.210] Warmup
[01:09:52.287] RPS: 1255312.63
[01:09:52.292] Benchmark
[01:10:08.337] RPS: 1273102.54
```

**Pull request merged**